Sources/Contour/Configurator/AppConfigConfigurator.cs

```diff
@@ -350,9 +354,42 @@ public IBusConfigurator Configure(string endpointName, IBusConfigurator cfg)
                 }
             }
 
+            var collectorType = endpointConfig.Metrics?.Collector;
+            if (!string.IsNullOrEmpty(collectorType))
+            {
+                try
+                {
+                    var metricsCollector = this.ResolveMetricsCollector(collectorType);
+                    cfg.CollectMetrics(metricsCollector);
+                }
+                catch (Exception e)
+                {
+                    Log.Warn(m => m("Could not load metric collector '{0}'", collectorType), e);
+                }
+            }
+            else
+            {
+                Log.Trace(m => m("Metric collector is not specified"));
+            }
+
             return cfg;
         }
 
+        private IMetricsCollector ResolveMetricsCollector(string collectorType)

```

Comment:
Contour use following logic: get dependency from dependencyresolver if specified or use default behavior if not. 
We do not use Activator.